File: app/src/utils/network.ts
import os from 'os';

/**
 * Get the local IP address
 * @returns string | null
 */
export function getLocalIPAddress(): string | null
{
	// Get the network interfaces
	const networkInterfaces = os.networkInterfaces();
	// Iterate over the network interfaces
	for(const interfaceName in networkInterfaces)
	{
		// Get the network information
		const networkInfo = networkInterfaces[interfaceName];
		// Check if the network information is valid
		if(networkInfo)
		{
			// Iterate over the network information
			for(const info of networkInfo)
			{
				// Check if the network information is an IPv4 address and not internal
				if(info.family === 'IPv4' && !info.internal)
					return info.address;
			}
		}
	}
	// Return null if no valid IP address is found
	return null;
}

File: ble/utils/network.py
#!/usr/bin/env python3
import socket
import psutil

def get_local_ip_address():
    """
    Get the local IP address.
    Iterates over all network interfaces and returns the first non-internal IPv4 address.
    Returns None if no valid IP address is found.
    """
    interfaces = psutil.net_if_addrs()
    for interface_name, addresses in interfaces.items():
        for addr in addresses:
            # Check for an IPv4 address that is not a loopback address.
            if addr.family == socket.AF_INET and not addr.address.startswith("127."):
                return addr.address
    return None
